### PR TITLE
RavenDB-8044

### PR DESCRIPTION
**Pull request by arekpalinski**

### CODE REVIEW

File: src/Raven.Client/Http/RequestExecutor.cs [hunk 1]
```diff
@@ -803,6 +803,14 @@ private HttpRequestMessage CreateRequest<TResult>(ServerNode node, RavenCommand<
                     throw new AuthorizationException("Forbidden access to " + chosenNode.Database + "@" + chosenNode.Url + ", " +
                         (Certificate == null ? "a certificate is required." : Certificate.FriendlyName + " does not have permission to access it or is unknown.") +
                         request.Method + " " + request.RequestUri);
+                case HttpStatusCode.Gone: // request not relevant for the chosen node - the database has been moved to a different one
+                    if (shouldRetry == false)
+                        return false;
+
+                    await UpdateTopologyAsync(chosenNode, Timeout.Infinite, forceUpdate: true).ConfigureAwait(false);
```
Comment (arekpalinski) [inline src/Raven.Client/Http/RequestExecutor.cs:810]: @ayende 
The reason I had to force topology update here was that both typologies, the previous and just retrieved ones, have `Topology.Etag == -1` so the new one was filtered out here:
https://github.com/ravendb/ravendb/blob/v4.0/src/Raven.Client/Http/NodeSelector.cs#L54

File: test/SlowTests/Issues/RavenDB_8044.cs [hunk 2]
```diff
@@ -0,0 +1,128 @@
...
+            }
+        }
+
+        private async Task DeleteDatabaseInCluster(string databaseName, IDocumentStore store, string fromNode)
+        {
+            var databaseResult = await store.Admin.Server.SendAsync(new DeleteDatabaseOperation(databaseName, true, fromNode));
+
+            foreach (var server in Servers)
+            {
+                await server.ServerStore.Cluster.WaitForIndexNotification(databaseResult.RaftCommandIndex + 1);
```
Comment (arekpalinski) [inline test/SlowTests/Issues/RavenDB_8044.cs:124]: @ayende
The need to increment the index value by `1` to make things working was tricky. I found it in a different test (https://github.com/ravendb/ravendb/blob/v4.0/test/RachisTests/Cluster.cs#L58).

Cannot we handle it better?

Comment (arekpalinski) [reply test/SlowTests/Issues/RavenDB_8044.cs:124]: It appears to be related to `DatabaseRecord.DeletionInProgress`. Is it?

Comment (karmeli87) [reply test/SlowTests/Issues/RavenDB_8044.cs:124]: It is. When you issue a deletion command it move the nodes to the DeletionInProgress Dictionary and then every node which delete the actual data issue another raft command to remove himself from the dictionary. And only after all nodes are removed from this dictionary the record is deleted

Comment (arekpalinski) [reply test/SlowTests/Issues/RavenDB_8044.cs:124]: OK, I see. That means `DeleteDatabaseOperation` isn't able to return a raft index which that will guarantee that db is deleted from all nodes, right?

Comment (talweiss1982) [reply test/SlowTests/Issues/RavenDB_8044.cs:124]: that is right

**Pull request merged**